### PR TITLE
[NTGDI] Fix MaskBlt error code and check pattern usage

### PR DESCRIPTION
**Pull request by katahiromz**
## Purpose

Follow-up to #7074. Fix `MaskBlt` behaviour.
JIRA issue: [CORE-13133](https://jira.reactos.org/browse/CORE-13133)

## Proposed changes

- Fix `MaskBlt` error code.
- Fix `MaskBlt` pattern usage check.

## TODO

- [x] Do small tests.
- [x] Do big tests.

## Comparison

BEFORE:
before
There were some failures.

AFTER:
after
Successful.

### CODE REVIEW

File: win32ss/gdi/ntgdi/bitblt.c [hunk 1]
```diff
@@ -337,18 +337,22 @@ NtGdiMaskBlt(
...
+    if (!hdcDest)
     {
         EngSetLastError(ERROR_INVALID_PARAMETER);
         return FALSE;
     }
 
+    UsesSource = ROP4_USES_SOURCE(rop4);
+    UsesPattern = ROP4_USES_PATTERN(rop4);
+    if (!hdcSrc && (UsesSource || UsesPattern))
+        return FALSE;
```
Comment (HBelusca) [inline win32ss/gdi/ntgdi/bitblt.c:354]: So here, there's no last-error being set?

**Review by HBelusca — APPROVED** [review]
Looks reasonable.

File: win32ss/gdi/ntgdi/bitblt.c (hunk 1 above)
Comment (katahiromz) [reply win32ss/gdi/ntgdi/bitblt.c:354]: There is no.

**Comment by katahiromz** [comment]
Test KVM (BEFORE): 
Test KVM (AFTER): aa27f87
Test VBox (BEFORE): 
Test VBox (AFTER): aa27f87

**Pull request merged**